src/lib.rs

```diff
@@ -25,11 +25,28 @@ struct DeriveWhere {
 
 impl Parse for DeriveWhere {
     fn parse(input: ParseStream) -> Result<Self> {
-        let bounds = Punctuated::<TraitBound, Token![,]>::parse_separated_nonempty(input)?
-            .into_iter()
-            .collect();
+        let found_semicolon = input.step(|cursor| {
+            for tree in cursor.token_stream() {
+                if let TokenTree::Punct(punct) = tree {
+                    if punct.as_char() == ';' {
+                        return Ok((true, *cursor));
+                    }
+                }
+            }
+
+            Ok((false, *cursor))
+        })?;
 
-        <Token![;]>::parse(input)?;
+        let bounds = if found_semicolon {
+            let bounds = Punctuated::<TraitBound, Token![,]>::parse_separated_nonempty(input)?
+                .into_iter()
+                .collect();
+            <Token![;]>::parse(input)?;
+
+            bounds
+        } else {
+            Vec::new()
+        };
```

Comment:
This new implementation is still up for discussion. @ModProg feel free to overwrite it.

Reply:
I would go with fork and advance_to, but let's create an issue about benchmarking this.

Reply:
Solved in another PR.